### PR TITLE
docs(Fusion): add Handling Events article to Fusion / Application section

### LINKED ISSUE #215: Add an article on handling events for Fusion
There should be an article in the Fusion / Creating an Application section documenting the recommended way of handling UI events in a Fusion application view.

It should:

- Have a basic code example of a declarative event listener on `<vaadin-button>` click, using a `private` view class method handler.
- Suggest where to find supported events in Vaadin Components and web platform docs.
- Mention using `async` in event listeners, for example, for accessing endpoints.
- Suggest things to do in event listeners, such as updating a managed application state, or view properties, or submitting a form. Use links to existing articles where appropriate.

The [LitElement’s Events article](https://lit-element.polymer-project.org/guide/events) provides a lot of ways for handling events, so it could be hard for beginners to use. We could link that in the article for more information.

### PR DESCRIPTION
**Pull request by platosha**
Fixes #215

### CODE REVIEW

**Review by haijian-vaadin — COMMENTED** [review]
There is a minor Vale error that could be fixed.
Personally, would prefer to eliminate the empty View interface to make the examples simpler.
Are you still working on more content or should be marked as ready for review?

**Comment by platosha** [comment]
> There is a minor Vale error that could be fixed.

Done.

> Personally, would prefer to eliminate the empty View interface to make the examples simpler.

Done.

> Are you still working on more content or should be marked as ready for review?

Marked as ready.

**Comment by jouni** [comment]
Can it be? An article without any Vale warnings or suggestions?! 🙀 

If that is the case, well done! 🥇 😄

**Review by haijian-vaadin — APPROVED** [review]

**Comment by haijian-vaadin** [comment]
Could you @jouni help to close merge the PR? The doc looks good, just some unrelated Vale errors from the base branch.

**Comment by jouni** [comment]
Yeah, sure, I’ll take a look. I just noticed that the preview deployment hasn’t actually been successful. I’ll see what fails.

File: articles/fusion/application/events.asciidoc [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+- `event.detail` in `CustomEvent` types, which is often used by web components for event-specific data, for example, `event.detail.value` is often used to propagate the new value in `property-changed` events and such
+- `event.preventDefault()` to cancel the built-in handling of the particular event, for example, to prevent the browser from navigating when the user clicks a link, after that click was already handled in the listener
+
+In the example below, the `event.detail.value` is used in the
+`value-changed` event handler to extract the edited `value` property of the
+`<vaadin-text-field>` element:
+
+[source,typescript]
+----
+include::{root}/frontend/demo/fusion/application/events/value-changed-view.ts[render,tags=snippet,indent=0,group=TypeScript]
```
Comment (jouni) [inline articles/fusion/application/events.asciidoc:57]:
```suggestion
include::{root}/frontend/demo/fusion/application/events/value-changed-view.ts[render,tags=snippet,indent=0]
```

**Review by jouni — CHANGES_REQUESTED** [review]
Build error:

```
9:12:06 PM: error Generating JavaScript bundles failed
9:12:06 PM: Can't resolve '../../init' in '/opt/build/repo/frontend/demo/fusion/application/events'
9:12:06 PM: If you're trying to use a package make sure that '../../init' is installed. If you're trying to use a local file make sure that the path is correct.
9:12:06 PM: error Generating JavaScript bundles failed
9:12:06 PM: Can't resolve '../../init' in '/opt/build/repo/frontend/demo/fusion/application/events'
9:12:06 PM: If you're trying to use a package make sure that '../../init' is installed. If you're trying to use a local file make sure that the path is correct.
```

**Review by jouni — APPROVED** [review]

**Review by jouni — CHANGES_REQUESTED** [review]
Oh, the build is still failing. That needs to be fixed first.

**Comment by haijian-vaadin** [comment]
The second example only shows me the Label, I don't see the text field
Screenshot 2021-03-11 at 13 33 12

**Review by haijian-vaadin — APPROVED** [review]
now the demos on the page also work, really nice, good job

**Comment by haijian-vaadin** [comment]
@jouni, now the build passes, can you help to take another look?

**Review by jouni — APPROVED** [review]

File: articles/fusion/application/events.asciidoc [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+To handle events, applications register event listener methods for specific event types (which denote the type of interaction, such as `click`) on the interactive targets (typically interface elements).
+
+== Declarative Event Listeners
+
+For registering event listeners in Fusion application views, declare them in your Lit templates using `@event` bindings with view class methods.
+In the example below, the view registers an event listener for the `sayHello`
+method to show the notification when the user clicks the button.
+
+[source,typescript]
```
Comment (jouni) [inline articles/fusion/application/events.asciidoc:21]:
```suggestion
[source,typescript, role=show-code]
```

File: articles/fusion/application/events.asciidoc [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+- `event.type` the event type string, for example, `click`
+- `event.target` the reference to the target (element) of the interaction
+- `event.detail` in `CustomEvent` types, which is often used by web components for event-specific data, for example, `event.detail.value` is often used to propagate the new value in `property-changed` events and such
+- `event.preventDefault()` to cancel the built-in handling of the particular event, for example, to prevent the browser from navigating when the user clicks a link, after that click was already handled in the listener
+
+In the example below, the `event.detail.value` is used in the
+`value-changed` event handler to extract the edited `value` property of the
+`<vaadin-text-field>` element:
+
+[source,typescript]
```
Comment (jouni) [inline articles/fusion/application/events.asciidoc:55]:
```suggestion
[source,typescript, role=show-code]
```

**Review by jouni — COMMENTED** [review]
I assume we would want to show the example code immediately without the need to click “Show code”.

**Pull request merged**